File: java2csharp/csharp/OpenTok/Sample.OpenTok.UI.Widget.TextChat/TextChatActivity.cs
namespace com.opentok.android.ui.textchat.sample
{

	using FragmentTransaction = android.app.FragmentTransaction;
	using NotificationManager = android.app.NotificationManager;
	using ServiceConnection = android.content.ServiceConnection;
	using Bundle = android.os.Bundle;
	using FragmentActivity = android.support.v4.app.FragmentActivity;
	using NotificationCompat = android.support.v4.app.NotificationCompat;
	using Log = android.util.Log;
	using MenuItem = android.view.MenuItem;
	using View = android.view.View;
	using ProgressBar = android.widget.ProgressBar;
	using RelativeLayout = android.widget.RelativeLayout;

	using ChatMessage = com.opentok.android.ui.textchat.widget.ChatMessage;
	using TextChatFragment = com.opentok.android.ui.textchat.widget.TextChatFragment;

	public class TextChatActivity : FragmentActivity, Session.SignalListener, Session.SessionListener, TextChatFragment.TextChatListener
	{

		private const string LOGTAG = "demo-text-chat";
		private const string SIGNAL_TYPE = "TextChat";

		// Replace with a generated Session ID
		public const string SESSION_ID = "";
		// Replace with a generated token (from the dashboard or using an OpenTok server SDK)
		public const string TOKEN = "";
		// Replace with your OpenTok API key
		public const string API_KEY = "";

		private Session mSession;

		private ProgressBar mLoadingBar;

		private bool resumeHasRun = false;

		private bool mIsBound = false;
		private NotificationCompat.Builder mNotifyBuilder;
		private NotificationManager mNotificationManager;
		private ServiceConnection mConnection;

		private RelativeLayout mTextChatViewContainer;
		private TextChatFragment mTextChatFragment;

		private FragmentTransaction mFragmentTransaction;

		private bool msgError = false;

		protected internal override void onCreate(Bundle savedInstanceState)
		{
			base.onCreate(savedInstanceState);
			ContentView = R.layout.layout_textchat_activity;

			mLoadingBar = (ProgressBar) findViewById(R.id.load_spinner);
			mLoadingBar.Visibility = View.VISIBLE;

			sessionConnect();
		}

		public override bool onOptionsItemSelected(MenuItem item)
		{
			switch (item.ItemId)
			{
				case android.R.id.home:
					onBackPressed();
					return true;
				default:
					return base.onOptionsItemSelected(item);
			}
		}

		public override void onPause()
		{
			base.onPause();

			if (mSession != null)
			{
				mSession.onPause();
			}
		}

		public override void onResume()
		{
			base.onResume();
			if (mSession != null)
			{
				mSession.onResume();
			}
		}

		public override void onStop()
		{
			base.onStop();
			if (Finishing && mSession != null)
			{
				mSession.disconnect();
			}
		}

		public override void onDestroy()
		{
			if (mSession != null)
			{
				mSession.disconnect();
			}
			base.onDestroy();
			finish();
		}

		public override void onBackPressed()
		{
			if (mSession != null)
			{
				mSession.disconnect();
			}
			base.onBackPressed();
		}

		private void sessionConnect()
		{
			if (mSession == null)
			{
				mSession = new Session(TextChatActivity.this, API_KEY, SESSION_ID);
				mSession.SignalListener = this;
				mSession.SessionListener = this;
				mSession.connect(TOKEN);
			}
		}

		// Initialize a TextChatFragment instance and add it to the UI
		private void loadTextChatFragment()
		{
			int containerId = R.id.fragment_textchat_container;
			mFragmentTransaction = FragmentManager.beginTransaction();
			mTextChatFragment = (TextChatFragment)this.FragmentManager.findFragmentByTag("TextChatFragment");

			if (mTextChatFragment == null)
			{
				mTextChatFragment = new TextChatFragment();
				mTextChatFragment.MaxTextLength = 1050;
				mTextChatFragment.setTextChatListener(this);
				mTextChatFragment.setSenderInfo(mSession.Connection.ConnectionId, mSession.Connection.Data);

				mFragmentTransaction.add(containerId, mTextChatFragment, "TextChatFragment").commit();
			}
		}

		public virtual bool onMessageReadyToSend(ChatMessage msg)
		{
			Log.d(LOGTAG, "TextChat listener: onMessageReadyToSend: " + msg.Text);

			if (mSession != null)
			{
				mSession.sendSignal(SIGNAL_TYPE, msg.Text);
			}
			return msgError;
		}

		public override void onSignalReceived(Session session, string type, string data, Connection connection)
		{
			Log.d(LOGTAG, "onSignalReceived. Type: " + type + " data: " + data);
			ChatMessage msg = null;
			if (!connection.ConnectionId.Equals(mSession.Connection.ConnectionId))
			{
				// The signal was sent from another participant. The sender ID is set to the sender's
				// connection ID. The sender alias is the value added as connection data when you
				// created the user's token.
				msg = new ChatMessage(connection.ConnectionId, connection.Data, data);
				// Add the new ChatMessage to the text-chat component
				mTextChatFragment.addMessage(msg);
			}
		}


		public override void onConnected(Session session)
		{
			Log.d(LOGTAG, "The session is connected.");

			mLoadingBar.Visibility = View.GONE;
			//loading text-chat ui component
			loadTextChatFragment();
		}

		public override void onDisconnected(Session session)
		{
			Log.d(LOGTAG, "The session disconnected.");
		}

		public override void onError(Session session, OpentokError opentokError)
		{
			Log.d(LOGTAG, "Session error. OpenTokError: " + opentokError.ErrorCode + " - " + opentokError.Message);
			OpentokError.ErrorCode errorCode = opentokError.ErrorCode;
			msgError = true;
		}

		public override void onStreamReceived(Session session, Stream stream)
		{
		}

		public override void onStreamDropped(Session session, Stream stream)
		{
		}

	}

}

File: java2csharp/java/OpenTok/old/OpenTok.Sample.UI.Widget.TextChat/TextChatActivity.java
package com.opentok.android.ui.textchat.sample;

import android.app.FragmentTransaction;
import android.app.NotificationManager;
import android.content.ServiceConnection;
import android.os.Bundle;
import android.support.v4.app.FragmentActivity;
import android.support.v4.app.NotificationCompat;
import android.util.Log;
import android.view.MenuItem;
import android.view.View;
import android.widget.ProgressBar;
import android.widget.RelativeLayout;

import com.opentok.android.Connection;
import com.opentok.android.OpentokError;
import com.opentok.android.Session;
import com.opentok.android.Stream;
import com.opentok.android.ui.textchat.widget.ChatMessage;
import com.opentok.android.ui.textchat.widget.TextChatFragment;

public class TextChatActivity extends FragmentActivity implements Session.SignalListener, Session.SessionListener, TextChatFragment.TextChatListener {

    private static final String LOGTAG = "demo-text-chat";
    private static final String SIGNAL_TYPE = "TextChat";

    // Replace with a generated Session ID
    public static final String SESSION_ID = "";
    // Replace with a generated token (from the dashboard or using an OpenTok server SDK)
    public static final String TOKEN = "";
    // Replace with your OpenTok API key
    public static final String API_KEY = "";

    private Session mSession;

    private ProgressBar mLoadingBar;

    private boolean resumeHasRun = false;

    private boolean mIsBound = false;
    private NotificationCompat.Builder mNotifyBuilder;
    private NotificationManager mNotificationManager;
    private ServiceConnection mConnection;

    private RelativeLayout mTextChatViewContainer;
    private TextChatFragment mTextChatFragment;

    private FragmentTransaction mFragmentTransaction;

    private boolean msgError = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.layout_textchat_activity);

        mLoadingBar = (ProgressBar) findViewById(R.id.load_spinner);
        mLoadingBar.setVisibility(View.VISIBLE);

        sessionConnect();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                onBackPressed();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    @Override
    public void onPause() {
        super.onPause();

        if (mSession != null) {
            mSession.onPause();
        }
    }

    @Override
    public void onResume() {
        super.onResume();
        if (mSession != null) {
            mSession.onResume();
        }
    }

    @Override
    public void onStop() {
        super.onStop();
        if (isFinishing() && mSession != null) {
            mSession.disconnect();
        }
    }

    @Override
    public void onDestroy() {
        if (mSession != null) {
            mSession.disconnect();
        }
        super.onDestroy();
        finish();
    }

    @Override
    public void onBackPressed() {
        if (mSession != null) {
            mSession.disconnect();
        }
        super.onBackPressed();
    }

    private void sessionConnect() {
        if (mSession == null) {
            mSession = new Session(TextChatActivity.this,
                    API_KEY, SESSION_ID);
            mSession.setSignalListener(this);
            mSession.setSessionListener(this);
            mSession.connect(TOKEN);
        }
    }

    // Initialize a TextChatFragment instance and add it to the UI
    private void loadTextChatFragment(){
        int containerId = R.id.fragment_textchat_container;
        mFragmentTransaction = getFragmentManager().beginTransaction();
        mTextChatFragment = (TextChatFragment)this.getFragmentManager().findFragmentByTag("TextChatFragment");

        if (mTextChatFragment == null) {
            mTextChatFragment = new TextChatFragment();
            mTextChatFragment.setMaxTextLength(1050);
            mTextChatFragment.setTextChatListener(this);
            mTextChatFragment.setSenderInfo(mSession.getConnection().getConnectionId(), mSession.getConnection().getData());

            mFragmentTransaction.add(containerId, mTextChatFragment, "TextChatFragment").commit();
        }
    }

    @Override
    public boolean onMessageReadyToSend(ChatMessage msg) {
        Log.d(LOGTAG, "TextChat listener: onMessageReadyToSend: " + msg.getText());

        if (mSession != null) {
            mSession.sendSignal(SIGNAL_TYPE, msg.getText());
        }
        return msgError;
    }

    @Override
    public void onSignalReceived(Session session, String type, String data, Connection connection) {
        Log.d(LOGTAG, "onSignalReceived. Type: " + type + " data: " + data);
        ChatMessage msg = null;
        if (!connection.getConnectionId().equals(mSession.getConnection().getConnectionId())) {
            // The signal was sent from another participant. The sender ID is set to the sender's
            // connection ID. The sender alias is the value added as connection data when you
            // created the user's token.
            msg = new ChatMessage(connection.getConnectionId(), connection.getData(), data);
            // Add the new ChatMessage to the text-chat component
            mTextChatFragment.addMessage(msg);
        }
    }


    @Override
    public void onConnected(Session session) {
        Log.d(LOGTAG, "The session is connected.");

        mLoadingBar.setVisibility(View.GONE);
        //loading text-chat ui component
        loadTextChatFragment();
    }

    @Override
    public void onDisconnected(Session session) {
        Log.d(LOGTAG, "The session disconnected.");
    }

    @Override
    public void onError(Session session, OpentokError opentokError) {
        Log.d(LOGTAG, "Session error. OpenTokError: " + opentokError.getErrorCode() + " - " + opentokError.getMessage());
        OpentokError.ErrorCode errorCode = opentokError.getErrorCode();
        msgError = true;
    }

    @Override
    public void onStreamReceived(Session session, Stream stream) {
    }

    @Override
    public void onStreamDropped(Session session, Stream stream) {
    }

}
